src/transforms/sample.jl

```diff
@@ -0,0 +1,72 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    Sample([rng], [wv::AbstractWeights], n; replace=true, ordered=false)
+
+Sample table rows by forwarding arguments to the `sample` function from StatsBase.jl.
+Optionally, a random number generator `rng` and a weight vector `wv` can be used.
+
+# Examples
+
+```julia
+Sample(1_000)
+Sample(1_000, replace=false)
+Sample(1_000, replace=false, ordered=true)
+
+# with rng
+using Random
+rng = MersenneTwister(2)
+Sample(rng, 1_000)
+
+# with weights
+using StatsBase
+wv = pweights([0.2, 0.1, 0.3])
+Sample(wv, 1_000)
+Sample(rng, wv, 1_000)
+```
+"""
+struct Sample{R<:AbstractRNG,W} <: Stateless
+  rng::R
+  wv::W
+  n::Int
+  replace::Bool
+  ordered::Bool
+end
+
+Sample(rng::AbstractRNG, wv::AbstractWeights, n::Int; replace=true, ordered=false) =
+  Sample(rng, wv, n, replace, ordered)
+
+Sample(rng::AbstractRNG, n::Int; replace=true, ordered=false) =
+  Sample(rng, nothing, n, replace, ordered)
+
+Sample(wv::AbstractWeights, n::Int; replace=true, ordered=false) =
+  Sample(Random.GLOBAL_RNG, wv, n, replace, ordered)
+
+Sample(n::Int; replace=true, ordered=false) =
+  Sample(Random.GLOBAL_RNG, nothing, n, replace, ordered)
```

Comment:
I wonder if we could just create a Sample(args; kwargs) and forward the args and kwargs to sample. That would make the code less verbose and less dependent on AbstractRNG and AbstractWeights.

Reply:
Maybe that is not possible because we need to pass the rows as the second argument to the function sample right?

Reply:
Yes, and in each method, the Sample builder has different default values. Also, type annotations bring better error messages.